packages/perspective-viewer-d3fc/src/js/legend/scrollableLegend.js

```diff
@@ -11,22 +11,39 @@ import {rebindAll} from "d3fc";
 import {getOrCreateElement} from "../utils/utils";
 import legendControlsTemplate from "../../html/legend-controls.html";
 import {cropCellContents} from "./styling/cropCellContents";
+import {draggableComponent} from "./styling/draggableComponent";
+import {resizableComponent} from "./styling/resizableComponent";
+
+const averageCellHeightPx = 16;
+const controlsHeightPx = 20;
 
 export default (fromLegend, settings) => {
     const legend = fromLegend || d3Legend.legendColor();
+
+    let domain = [];
     let pageCount = 1;
-    let pageSize = 15;
+    let pageSize;
     let pageIndex = settings.legend ? settings.legend.pageIndex : 0;
     let decorate = () => {};
+    let draggable = draggableComponent();
+    let resizable;
 
     const scrollableLegend = selection => {
-        const domain = legend.scale().domain();
-        pageCount = Math.ceil(domain.length / pageSize);
+        domain = legend.scale().domain();
+        pageCount = calculatePageCount();

```

Comment:
Can this line be dropped? `pageCount` gets set again inside `calculatePageSize`. The initial value is likely to be undefined anyway because `pageIndex` hasn't been set yet.

Reply:
Thanks 💯